### PR TITLE
Adds support for HTTP2 via axum.

### LINKED ISSUE #16: HTTP2 Support
I was wondering if the maintainers can add HTTP2 support as a feature request. So it can be worked later to be added to the framework.
Thanks.

### PR DESCRIPTION
**Pull request by novemberkilo**
Closes #16 

// @houqp

### CODE REVIEW

File: roapi-http/Cargo.toml [hunk 1]
```diff
@@ -21,8 +21,8 @@ snmalloc-rs = { version = "0.2", optional = true }
 
 # dependencies related to axum
 tokio = { version = "1", features = ["rt-multi-thread"] }
-hyper = { version = "0", features = ["http1", "server", "stream", "runtime"] }
-axum = "0.3"
+hyper = { version = "0", features = ["http1", "http2", "server", "stream", "runtime"] }
```
Comment (novemberkilo) [inline roapi-http/Cargo.toml:24]: I'm not :100: about this change. Adding `http2` to  the features of `axum` [results in exactly this](https://github.com/tokio-rs/axum/blob/254d8fde17555cb526f42052d23ffd4955284e98/axum/Cargo.toml#L16) so perhaps adding to `hyper`'s features is redundant?

Comment (houqp) [reply roapi-http/Cargo.toml:24]: yes, features are additive in cargo, so if it's turned on in axum, it will be transitively turned for for hyper by axum.

**Comment by houqp** [comment]
Thanks @novemberkilo , is the feature gate all we need to turn on this feature? I am guessing we still only need to bind to a single port and the client will negotiate with the server to use either HTTP or HTTP2?

**Comment by novemberkilo** [comment]
> I am guessing we still only need to bind to a single port and the client will negotiate with the server to use either HTTP or HTTP2?

I'll check the `hyper` docs and also do some ad-hoc testing with `curl` locally. I'll report back.

**Comment by novemberkilo** [comment]
@houqp locally testing with curl is not yielding anything definitive. However, I see 

> And users who want HTTP/2 can opt-in, knowing they need a newer compiler version, but setting hyper = { version = "0.11", features = ["http2"] }.
Besides opting in, there wouldn't be a need to do anything else: the client and server would both be able to speak HTTP/2 automatically.

From the issue commentary for when `HTTP/2` was added to hyper https://github.com/hyperium/hyper/issues/304#issuecomment-357338916

I've also got one of the contributors making this assertion [here](https://www.reddit.com/r/rust/comments/8t6rme/comment/e15bmy9/?utm_source=share&utm_medium=web2x&context=3)

File: roapi-http/Cargo.toml (hunk 1 above)
Comment (novemberkilo) [reply roapi-http/Cargo.toml]: Ok I have removed the redundant addition of the feature to `hyper`

**Review by houqp — APPROVED** [review]

**Comment by houqp** [comment]
Thanks @novemberkilo for the deepdive on this :) I filed https://github.com/roapi/roapi/issues/120 as a follow up issue.

**Pull request merged**